code/modules/ninja/energy_katana.dm

```diff
@@ -1,30 +1,96 @@
-/obj/item/weapon/katana/energy
+/obj/item/weapon/dash
+	name = "abstract dash weapon"
+	var/max_charges = 3
+	var/current_charges = 3
+	var/charge_rate = 30 //In deciseconds
+	var/dash_toggled = TRUE
+
+	var/bypass_density = FALSE //Can we beam past windows/airlocks/etc
+
+	var/start_effect_type = /obj/effect/temp_visual/dir_setting/ninja/phase/out
+	var/end_effect_type = /obj/effect/temp_visual/dir_setting/ninja/phase
+	var/beam_icon_state = "blur"
+	var/dash_beam_type = /obj/effect/ebeam
+
+/obj/item/weapon/dash/proc/charge()
+	current_charges = Clamp(current_charges + 1, 0, max_charges)
+	if(istype(loc, /mob/living))
+		to_chat(loc, "<span class='notice'>[src] now has [current_charges]/[max_charges] charges.</span>")
+
+/obj/item/weapon/dash/attack_self(mob/user)
+	dash_toggled = !dash_toggled
+	if(dash_toggled)
+		to_chat(user, "<span class='notice'>[src] is now in dash mode.</span>")
```

Comment:
Replace the if/else with `to_chat(user, "<span class='notice'>You [dash_toggle ? "enable" : "disable"] the dash function on [src].</span>")`